internal/x509/cert.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package x509
+
+import (
+	"bytes"
+	"crypto/x509"
+)
+
+// IsRootCertificate returns true if cert is a root certificate.
+// A root certificate MUST be a self-signed and self-issued CA certificate with
+// valid BasicConstraints.
+func IsRootCertificate(cert *x509.Certificate) (bool, error) {
+	// CheckSignatureFrom also checks cert.BasicConstraintsValid
+	if err := cert.CheckSignatureFrom(cert); err != nil {
+		return false, err
+	}
+	return cert.IsCA && bytes.Equal(cert.RawSubject, cert.RawIssuer), nil
```

Comment:
Is `cert.IsCA` required? Can it be just a self-signed certificate? Is there any spec reference?

Reply:
Yeah, spec: https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#root-and-intermediate-ca-certificates

Basically, we require root and intermediate certs to be CA issued:
`The cA field MUST be set to true`

Reply:
The spec states
> Valid certificate chain MUST contain a root certificate. If the certificate chain contains only a root certificate then the root certificate MUST meet [Leaf Certificates](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#leaf-certificates) requirements and ignore [Root and Intermediate CA Certificates](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#root-and-intermediate-ca-certificates) requirements.

Reply:
Yes, but do we want to support the self-signed case? This is an `internal` helper function for `--tsa-root-cert`. Based on our testing, none of the TSA is using a chain of length 1 so far.

Reply:
What about private TSAs? Of course, we can improve that in the next iteration.

Reply:
Updated.